File: aemet/Aemet-Java-IvanRC/src/main/java/validator/NombreFicheroValidator.java
package validator;

import java.io.FileNotFoundException;

public class NombreFicheroValidator {
    public static boolean validateNombreFichero(String nombre) throws FileNotFoundException {
        var regexNombreFicheroCsv = "Aemet[0-9]{8}.csv";
        if(!nombre.matches(regexNombreFicheroCsv)){
            throw new FileNotFoundException("Ese nombre de fichero no es valido, debe ser, como: Aemet20170101.csv");
        }
        return true;
    }
}

File: aemet/Aemet-kotlin-IvanRC/src/main/kotlin/validator/NombreFicheroValidator.kt
package validator

import java.io.FileNotFoundException

fun String.validate(): Boolean{
    val regexNombreFicheroCsv = Regex("Aemet[0-9]{8}.csv")
    require(this.matches(regexNombreFicheroCsv)){
        throw FileNotFoundException("Ese nombre de fichero no es valido, debe ser, como: Aemet20170101.csv")
    }
    return true
}